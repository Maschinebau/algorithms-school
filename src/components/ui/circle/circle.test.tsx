import { render } from "@testing-library/react"
import { Circle } from "./circle"
import { ElementStates } from "../../../types/element-states"

describe("Circle component snapshot", () => {
  it("renders circle without letters", () => {
    const { asFragment } = render(<Circle />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with letters", () => {
    const { asFragment } = render(<Circle letter={"hi"} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with head", () => {
    const { asFragment } = render(<Circle head={"head"} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with component in head", () => {
    const { asFragment } = render(<Circle head={<Circle />} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with tail", () => {
    const { asFragment } = render(<Circle tail={"head"} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with component in tail", () => {
    const { asFragment } = render(<Circle tail={<Circle />} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with index", () => {
    const { asFragment } = render(<Circle index={5} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with isSmall prop", () => {
    const { asFragment } = render(<Circle isSmall={true} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with default state", () => {
    const { asFragment } = render(<Circle state={ElementStates.Default} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with changing state", () => {
    const { asFragment } = render(<Circle state={ElementStates.Changing} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders circle with modified state", () => {
    const { asFragment } = render(<Circle state={ElementStates.Modified} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
