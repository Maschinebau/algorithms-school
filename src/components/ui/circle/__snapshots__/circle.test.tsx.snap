// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component snapshot renders circle with changing state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with component in head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with component in tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with default state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      head
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      5
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with isSmall prop 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with letters 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        hi
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with modified state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle with tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      head
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshot renders circle without letters 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;
